vortex-scalar/src/value.rs

```diff
@@ -14,19 +14,17 @@ use crate::pvalue::PValue;
 /// Note that these values can be deserialized from JSON or other formats. So a PValue may not
 /// have the correct width for what the DType expects. Primitive values should therefore be
 /// read using [crate::PrimitiveScalar] which will handle the conversion.
-#[derive(Debug, Clone, PartialEq, PartialOrd, Hash)]
+#[derive(Debug, Clone)]
 pub struct ScalarValue(pub(crate) InnerScalarValue);
 
-#[derive(Debug, Clone, PartialEq, PartialOrd, Hash)]
+#[derive(Debug, Clone)]
 pub(crate) enum InnerScalarValue {
+    Null,
     Bool(bool),
     Primitive(PValue),
     Buffer(Arc<ByteBuffer>),
     BufferString(Arc<BufferString>),
     List(Arc<[ScalarValue]>),
-    // It's significant that Null is last in this list. As a result generated PartialOrd sorts Scalar
```

Comment:
you should leave a comment

Reply:
What comment? ScalarValue has no ordering

Reply:
I think the comment should be roughly on Scalar that says that null ordering is dtype specific and then maybe explain that deferring to rust option ordering means nulls are lowest. Before all the ordering logic ended up here in scalarvalue

Reply:
Good point, added comment to Scalar doc